hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/IMapSqlConnector.java

```diff
@@ -126,13 +133,38 @@ public Table createTable(
         );
     }
 
-    @Override
-    public boolean supportsNestedLoopReader() {
-        return true;
+    @Nonnull @Override
+    public Vertex fullScanReader(
+            @Nonnull DAG dag,
+            @Nonnull Table table0,
+            @Nullable Expression<Boolean> predicate,
+            @Nonnull List<Expression<?>> projections
+    ) {
+        PartitionedMapTable table = (PartitionedMapTable) table0;
+
+        List<TableField> fields = table.getFields();
+        QueryPath[] paths = fields.stream().map(field -> ((MapTableField) field).getPath()).toArray(QueryPath[]::new);
+        QueryDataType[] types = fields.stream().map(TableField::getType).toArray(QueryDataType[]::new);
+
+        Vertex vStart = dag.newUniqueVertex(
+                toString(table),
+                SourceProcessors.readMapP(table.getMapName()));
```

Comment:
I'd suggest you to use new `OnHeapMapScanP`. WDYT?

Reply:
I tried to apply it I got `com.hazelcast.sql.HazelcastSqlException: "metaSupplier" must be serializable`
`Caused by: java.io.NotSerializableException: com.hazelcast.sql.impl.extract.QueryPath`

Reply:
Got you, will be fixed in my indexes PR.

Reply:
@Fly-Style you'll have to resolve the conflict in `IMapSqlConnector` when you'll be merging your PR :)